### ISSUE TITLE
"Stream has ended unexpectedly" trying to instantiate `PdfReader` with large PDF printed from Firefox

### ISSUE THREAD

**Issue by stevecj** [py-pdf/pypdf#2486]
Trying to instantiate `PdfReader` with a fairly large PDF printed from Firefox so I can scrape data from it and then manipulate the pages, I get "pypdf.errors.PdfStreamError: Stream has ended unexpectedly". The problem happens even when I supply the `strict=False` option.

The same problem does not occur with smaller files printed from Firefox. Small or large files printed from Firefox can be opened and viewed successfully in Firefox or LibreOffice.

## Environment

Which environment were you using when you encountered the problem?

```bash
$ python -m platform
Linux-5.15.133.1-microsoft-standard-WSL2-x86_64-with-glibc2.28

$ python -c "import pypdf;print(pypdf._debug_versions)"
pypdf==4.0.1, crypt_provider=('cryptography', '41.0.7'), PIL=10.2.0

$ firefox --version
Mozilla Firefox 115.6.0esr
```

## Code + PDF

This is a minimal, complete example that shows the issue:

```python
        pdf_reader = PdfReader(self.filename, strict=False)
```

Unfortunately, due to the nature of the problem, I have no _small_ file that I can attach that reproduces the problem. Here is a file that reproduces the problem though:
[mozilla.pdf](https://github.com/py-pdf/pypdf/files/14443303/mozilla.pdf)

## Traceback

This is the complete traceback I see:

```
  File "/home/stevjorg/work/hub2/django/dhbackbone/pdf_generator/generator.py", line 330, in _scrape_page_replacement_data_from_pdf
    pdf_reader = PdfReader(self.filename, strict=False)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/stevjorg/.pyenv/versions/3.11.7/lib/python3.11/site-packages/pypdf/_reader.py", line 318, in __init__
    self.read(stream)
  File "/home/stevjorg/.pyenv/versions/3.11.7/lib/python3.11/site-packages/pypdf/_reader.py", line 1335, in read
    self._find_eof_marker(stream)
  File "/home/stevjorg/.pyenv/versions/3.11.7/lib/python3.11/site-packages/pypdf/_reader.py", line 1406, in _find_eof_marker
    line = read_previous_line(stream)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/stevjorg/.pyenv/versions/3.11.7/lib/python3.11/site-packages/pypdf/_utils.py", line 285, in read_previous_line
    raise PdfStreamError(STREAM_TRUNCATED_PREMATURELY)
pypdf.errors.PdfStreamError: Stream has ended unexpectedly
[2024-02-28 20:26:45 PST] ERROR: django.server "POST /api/v1/ip_revisions/851/datasheet/ HTTP/1.1" 500 145; (basehttp.py:161) pr: MainProcess (1227518)
```

**Comment by stevecj** [comment]
After posting the above, I tried the same thing from the Python command line with the same file and did not get the error, which is obviously weird. In its original context, I had added a ridiculously excessive sleep after printing from Firefox assuming that would ensure that the file was completely written, but maybe the file was _not_ fully written until Firefox was later closed, or something like that?

I had also moved the file prior to trying to open it with `PdfReader`, but I guess that would not necessarily ensure that it was closed, and Firefox could still have retained a handle to the file while it was moved.
